devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -104,15 +105,18 @@ private void registerTasks(Project project) {
                     Task testNative = tasks.create(TEST_NATIVE_TASK_NAME, QuarkusTestNative.class);
                     testNative.dependsOn(buildNative);
                     testNative.setShouldRunAfter(Collections.singletonList(tasks.findByName(JavaPlugin.TEST_TASK_NAME)));
-                    tasks.withType(Test.class).forEach(t -> {
+
+                    Consumer<Test> configureTestTask = t -> {
                         // Quarkus test configuration task which should be executed before any Quarkus test
                         t.dependsOn(quarkusTestConfig);
                         // also make each task use the JUnit platform since it's the only supported test environment
                         t.useJUnitPlatform();
-                    });
+                    };
+                    tasks.withType(Test.class).forEach(configureTestTask);
+                    tasks.withType(Test.class).whenTaskAdded(t -> configureTestTask.accept(t));
```

Comment:
`tasks.withType(Test.class).whenTaskAdded(configureTestTask)` ?

Reply:
Can't. It takes an Action, which is not compatible with Consumer.